docs/content/configuration/historical.md

```diff
@@ -49,6 +49,7 @@ Druid uses Jetty to serve HTTP requests.
 |`druid.server.http.numThreads`|Number of threads for HTTP requests.|max(10, (Number of cores * 17) / 16 + 2) + 30|
 |`druid.server.http.maxIdleTime`|The Jetty max idle time for a connection.|PT5m|
 |`druid.server.http.defaultQueryTimeout`|Query timeout in millis, beyond which unfinished queries will be cancelled|300000|
+|`druid.server.http.maxQueryTimeout`|Maximum allowed value (in milliseconds) for `timeout` parameter. See [query-context](query-context.html) to know more about `timeout`. Query is rejected if the query context `timeout` is greater than this value. |Long.MAX_VALUE|
```

Comment:
I don't think the query should be rejected. The query should run with this maxTimeout value instead

Reply:
I choose to reject the query b/c:
1. To let user know about  max `timeout` value so that they can adjust the context.
2. Druid should not change the actual intent of the query, otherwise user might think query has been accepted but timing out before the context timeout, which is confusing in general.

Reply:
Rejecting the query sounds good.